### PR TITLE
Use longer lived lsp fee offers for redeeming

### PR DESCRIPTION
**Pull request by gcomte**
An LSP may offer multiple fee rates, tied to different expiration dates.

This PR exposes this election offer-election in some places and uses opinionated values in others.

### CODE REVIEW

File: src/lipalightninglib.udl [hunk 1]
```diff
@@ -326,6 +326,9 @@ interface Swap {
     [Throws=SwapError]
     SwapAddressInfo create(OpeningFeeParams? lsp_fee_params);
 
+    [Throws=LnError]
+    CalculateLspFeeResponse calculate_swap_lsp_fee_for_amount(u64 amount_sat);
```
Comment (danielgranhao) [inline src/lipalightninglib.udl:330]: Given we are within the swap interface, there is no need to have "swap" in the method name :) 

```suggestion
    CalculateLspFeeResponse calculate_lsp_fee_for_amount(u64 amount_sat);
```

File: src/lib.rs [hunk 2]
```diff
@@ -625,9 +625,13 @@ impl LightningNode {
 
     /// Calculate the actual LSP fee for the given amount of an incoming payment.
     /// If the already existing inbound capacity is enough, no new channel is required.
+    /// The LSP may offer multiple fee rates, tied to different expiration dates.
+    /// Increased expiry dates mean higher fee rates.
+    /// This method returns the best offer within the given expiry.
     ///
     /// Parameters:
     /// * `amount_sat` - amount in sats to compute LSP fee for
+    /// * `expiry` - expiry time in seconds
```
Comment (danielgranhao) [inline src/lib.rs:634]: The docs here are not up-to-date. We probably don't want any changes

File: src/onchain/channel_closes.rs [hunk 3]
```diff
@@ -239,7 +239,7 @@ impl ChannelClose {
 
         let lsp_fees = self
             .support
-            .calculate_lsp_fee_for_amount(send_amount_sats)
+            .calculate_lsp_fee_for_amount(send_amount_sats, None)
```
Comment (danielgranhao) [inline src/onchain/channel_closes.rs:242]: This logic needs to be fixed anyway (we should use lsp fee params provided as input -  let's fix in another PR), but I would use here the new swap version of the method.

File: src/onchain/swap.rs [hunk 4]
```diff
@@ -232,7 +234,7 @@ impl Swap {
 
         let lsp_fees = self
             .support
-            .calculate_lsp_fee_for_amount(send_amount_sats)?
+            .calculate_lsp_fee_for_amount(send_amount_sats, None)?
```
Comment (danielgranhao) [inline src/onchain/swap.rs:237]: This logic needs to be fixed anyway (we should use lsp fee params provided as input -  let's fix in another PR), but I would use here the new swap version of the method.

File: src/onchain/swap.rs [hunk 5]
```diff
@@ -282,6 +284,27 @@ impl Swap {
...
+    ) -> Result<CalculateLspFeeResponse> {
+        calculate_swap_lsp_fee_for_amount(&self.support, amount_sat)
+    }
+}
+
+pub fn calculate_swap_lsp_fee_for_amount(
+    support: &Support,
+    amount_sat: u64,
+) -> Result<CalculateLspFeeResponse> {
+    support.calculate_lsp_fee_for_amount(amount_sat, Some(TWO_WEEKS))
```
Comment (danielgranhao) [inline src/onchain/swap.rs:307]: Do we need this function? It seems to me that everywhere it is needed we have an instance of `Swap`, so we can simply use the method

File: src/lib.rs (hunk 2 above)
Comment (gcomte) [reply src/lib.rs]: Removed: 2a522c732a12c16c00c216b17e7199e9199a1819

File: src/onchain/channel_closes.rs (hunk 3 above)
Comment (gcomte) [reply src/onchain/channel_closes.rs:242]: I'm gonna create a second PR :+1:

Comment (gcomte) [reply src/onchain/channel_closes.rs]: Temporary fix: 9854933355983b8f7fc790547759b225ce1dc4eb

File: src/onchain/swap.rs (hunk 5 above)
Comment (gcomte) [reply src/onchain/swap.rs]: Not really. But I introduced `Swap` where it was missing: 5e420e731955c7579898a75f26761c19658991a0

File: src/lipalightninglib.udl (hunk 1 above)
Comment (gcomte) [reply src/lipalightninglib.udl]: d7e25c0e7143bb8e9864d23e47f4ea1804d1bd46

**Pull request merged**